### PR TITLE
Show used solver in `conda info` output.

### LINKED ISSUE #13265: Show configured solver in `conda info` call
### Checklist

- [X] I added a descriptive title
- [X] I searched open requests and couldn't find a duplicate

### What is the idea?

For debugging purposes we should list the configured solver in the output of `conda info`

### Why is this needed?

conda-libmamba-solver is about to become the new solver for conda, and we have a way to see that it's used by looking at the user-agent.

But: I assume we might see more solver to show up, that may or may not add their information to the user agent, and it would be better to just have the value of `context.solver` be printed with conda info.

### What should happen?

When running `conda info` (etc) it should list the configured solver backend: `solver : classic` or similar.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by jezdez**
Fix #13265.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- ~[ ] Add / update outdated documentation?~

### CODE REVIEW

**Comment by jezdez** [comment]
pre-commit.ci autofix

File: conda/base/context.py [hunk 1]
```diff
@@ -1005,6 +1005,21 @@ def log_level(self) -> int:
         else:
             return logging.WARNING  # 30
 
+    @memoizedproperty
+    def solver_user_agent(self):
```
Comment (dholth) [inline conda/base/context.py:1009]: Is this really not covered by the unit tests?
Could we skip @memoizedproperty on this one?

File: conda/base/context.py [hunk 2]
```diff
@@ -1005,6 +1005,21 @@ def log_level(self) -> int:
         else:
             return logging.WARNING  # 30
 
+    @memoizedproperty
```
Comment (jezdez) [inline conda/base/context.py:1008]:
```suggestion
```

File: conda/base/context.py [hunk 3]
```diff
@@ -1014,18 +1029,7 @@ def user_agent(self):
...
-                # Solver.user_agent has to be a static or class method
-                user_agent_str += f" {solver_backend.user_agent()}"
-            except Exception as exc:
-                log.debug(
-                    "User agent could not be fetched from solver class '%s'.",
-                    self.solver,
-                    exc_info=exc,
-                )
-            builder.append(user_agent_str)
+            builder.append(self.solver_user_agent)
```
Comment (jezdez) [inline conda/base/context.py:1032]:
```suggestion
            builder.append(self.solver_user_agent())
```

File: conda/cli/main_info.py [hunk 4]
```diff
@@ -242,6 +248,12 @@ def get_info_dict(system=False):
 
     active_prefix_name = env_name(context.active_prefix)
 
+    solver = {
+        "name": context.solver,
+        "user_agent": context.solver_user_agent,
```
Comment (jezdez) [inline conda/cli/main_info.py:253]:
```suggestion
        "user_agent": context.solver_user_agent(),
```

**Review by dholth — APPROVED** [review]

**Review by jaimergp — APPROVED** [review]
I had relied on the user agent (or `conda config`) for this kind of info til now, but I guess it doesn't hurt to have it as a separate field.

**Comment by jezdez** [comment]
> I had relied on the user agent (or `conda config`) for this kind of info til now, but I guess it doesn't hurt to have it as a separate field.

Same to be honest, but it doesn't hurt to be able to programmatically get it (via `conda info --json`) without having to parse the user agent.

**Pull request merged**